### ISSUE TITLE
Steam Deck support

### ISSUE THREAD

**Issue by RossComputerGuy** [ExpidusOS/file-manager#28]
We're looking to support the Steam Deck and need suggestions for features or improvements to this application to improve the experience. So far, we've been able to get the File Manager to open in the Steam Deck big picture but had to remap the D-pad to arrow keys and the A button to enter. First enhancement will likely be true gamepad support.

**Comment by ccarlile** [comment]
Hi there, I quite like the UI as opposed to Dolphin's which uses a lot of tooltips and trips up the window size in game mode. However, in certain views the contrast between text and background is so low the colors are almost identical. Installed onto steam deck via discover store today.

https://i.imgur.com/Nw97SU6.png

**Comment by RossComputerGuy** [comment]
@ccarlile Try the latest build, that should have the contrast fixed.

**Comment by ccarlile** [comment]
So I downloaded the .flatpak from the releases page here on github, but it doesn't seem to have helped. FWIW, I tried installing both by double-clicking the .flatpak and installing it through the software center, as well as using `flatpak install` directly. The software center identified the bundle as being published with version 0.2.1

**Comment by RossComputerGuy** [comment]
@ccarlile What version does the app say in about? Right now, the current commit 0.3.0 is on should have it fixed as libtokyo was updated with new fonts. Also, you should be able to download the tarball from GitHub Actions and execute that if the Flatpak isn't working as expected.

**Comment by ccarlile** [comment]
Yeah, I was able to find the "About" in the settings and the flatpak app and the linux .zip both seem to report 0.2.1

**Comment by RossComputerGuy** [comment]
@ccarlile Are you sure you downloaded the right version? If you downloaded the Zip from CI, it should report 0.3.0.

**Comment by ccarlile** [comment]
I downloaded from CI and the contrast issue is indeed fixed. Here's some more notes, roughly in order of desired:
- Is there a way to ascend the directory tree? I see the back button but not an "up" button. Maybe just including `..` as an option when viewing hidden files would be sufficient.
-  Single click shouldn't open the file IMO, in game mode this might mess up the windowing
  - Related: file previews would be cool
- I'd like to be able to remove entries from the sidebar (I don't really need the "Movies" folder on my deck)
- Some config mechanism would be nice to e.g. set our own colors or other preferences

**Comment by ccarlile** [comment]
And FWIW, proper gamepad support isn't super critical IMO on the deck - if launched through BPM, steam input is sufficient to emulate M&K desktop binds and indeed that's an available preset for the deck controls

**Comment by RossComputerGuy** [comment]
> Is there a way to ascend the directory tree? I see the back button but not an "up" button. Maybe just including .. as an option when viewing hidden files would be sufficient.

I was thinking the title in the app bar could work like breadcrumbs so that could be good.

> Single click shouldn't open the file IMO, in game mode this might mess up the windowing

How would you want the files to open? In what way does it mess up the windowing?

> Related: file previews would be cool

Planned

> I'd like to be able to remove entries from the sidebar (I don't really need the "Movies" folder on my deck)

Should be able to add a way to hide specific library entries, might add it in with 0.3.0

> Some config mechanism would be nice to e.g. set our own colors or other preferences

Application settings exists. However, we do plan on globalized theming on all of our apps with libtokyo and for it to update the themes with our DE. Not sure if it would be possible outside of ExpidusOS though. But there might be a way to support changing the colors in future versions of the libtokyo UI library.

> proper gamepad support isn't super critical IMO on the deck

I was thinking proper gamepad support could help since it seems Flutter doesn't translate the buttons right? Or it might be easier to change the controls in the input settings on the Steam Deck?
